### PR TITLE
buildah 1.27.3 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/buildah.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  head "https://github.com/containers/buildah.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "1"
+
+    ldflags = "-s -w"
+    ldflags << "-linkmode external -extldflags \"-static -lm\"" if OS.linux?
+    tags = "netgo osusergo exclude_graphdriver_btrfs exclude_graphdriver_devicemapper seccomp apparmor selinux"
```
Comment (SMillerDev) [inline Formula/buildah.rb:16]: Let's make it an array

**Comment by chenrui333** [comment]
```
/home/linuxbrew/.linuxbrew/Cellar/go/1.19.4/libexec/pkg/tool/linux_amd64/link: running gcc-11 failed: exit status 1
  /usr/bin/ld: /tmp/go-link-4255848032/000029.o: in function `New':
  /_/github.com/miekg/pkcs11@v1.1.1/tmp/buildah-20230102-23905-vwygfn/buildah-1.27.3/vendor/github.com/miekg/pkcs11/pkcs11.go:77: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
  /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib/libgpgme.a(data.o): in function `gpg_error_from_syserror':
  data.c:(.text+0xa5): undefined reference to `gpg_err_code_from_syserror'
  /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib/libgpgme.a(data.o): in function `_gpgme_data_release.part.0':
  data.c:(.text+0xd5): undefined reference to `gpgrt_lock_lock'
  /usr/bin/ld: data.c:(.text+0x111): undefined reference to `gpgrt_lock_unlock'
  /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib/libgpgme.a(data.o): in function `_gpgme_data_get_dserial':
  data.c:(.text+0x2[44](https://github.com/Homebrew/homebrew-core/actions/runs/3825459718/jobs/6508499998#step:7:45)): undefined reference to `gpgrt_lock_lock'
  /usr/bin/ld: data.c:(.text+0x276): undefined reference to `gpgrt_lock_unlock'
  /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib
```

**Comment by dawidd6** [comment]
Can we not use the upstream-provided Makefile?

**Comment by dawidd6** [comment]
We also apparently need `runc` for `buildah` to work properly:

image

**Comment by chenrui333** [comment]
> We also apparently need `runc` for `buildah` to work properly:
> 
> image

good point, I will add runc in another PR.

**Comment by ismail** [comment]
> ```
> /home/linuxbrew/.linuxbrew/Cellar/go/1.19.4/libexec/pkg/tool/linux_amd64/link: running gcc-11 failed: exit status 1
>   /usr/bin/ld: /tmp/go-link-4255848032/000029.o: in function `New':
>   /_/github.com/miekg/pkcs11@v1.1.1/tmp/buildah-20230102-23905-vwygfn/buildah-1.27.3/vendor/github.com/miekg/pkcs11/pkcs11.go:77: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
>   /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib/libgpgme.a(data.o): in function `gpg_error_from_syserror':
>   data.c:(.text+0xa5): undefined reference to `gpg_err_code_from_syserror'
>   /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib/libgpgme.a(data.o): in function `_gpgme_data_release.part.0':
>   data.c:(.text+0xd5): undefined reference to `gpgrt_lock_lock'
>   /usr/bin/ld: data.c:(.text+0x111): undefined reference to `gpgrt_lock_unlock'
>   /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib/libgpgme.a(data.o): in function `_gpgme_data_get_dserial':
>   data.c:(.text+0x2[44](https://github.com/Homebrew/homebrew-core/actions/runs/3825459718/jobs/6508499998#step:7:45)): undefined reference to `gpgrt_lock_lock'
>   /usr/bin/ld: data.c:(.text+0x276): undefined reference to `gpgrt_lock_unlock'
>   /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gpgme/1.18.0_1/lib
> ```

I don't know why it was done like this, but gpgme shouldn't be built as a static library: https://github.com/Homebrew/homebrew-core/blob/944855c15ea73ee344cc92baf87b1880ba10a70a/Formula/gpgme.rb#L66

**Comment by dawidd6** [comment]
We build both static and shared library of `gpgme`.

**Comment by ismail** [comment]
> We build both static and shared library of `gpgme`.

That’s a security problem. One security bug in gpgme and you’ll have to recompile everything that links to it statically. I don’t know what’s the actual gain from statical support here, but in general no Linux distro would do this.

**Comment by dawidd6** [comment]
I guess we do this to give users a choice, but we still aim to link everything dynamically. Not sure though.

**Comment by SMillerDev** [comment]
I'm sure at some point someone asked for it and we obliged because there isn't much of a reason not to do this. All our software is supposed to link dynamically so any recompiling is up to users.

**Comment by chenrui333** [comment]
closing it for now.

**Pull request closed without merging**